src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -1193,31 +1193,42 @@ public void deleteLocalProgrammingExerciseReposFolder(ProgrammingExercise progra
     }
 
     /**
-     * Zip the content of a git repository that contains a participation.
+     * Get the content of a git repository that contains a participation, as zip or directory.
      *
      * @param repo            Local Repository Object.
      * @param repositoryDir   path where the repo is located on disk
-     * @param hideStudentName option to hide the student name for the zip file
-     * @return path to zip file.
-     * @throws IOException if the zipping process failed.
+     * @param hideStudentName option to hide the student name for the zip file or directory
+     * @param zipOutput       If true the method returns a zip file otherwise a directory.
+     * @return path to zip file or directory.
+     * @throws IOException if the zipping or copying process failed.
      */
-    public Path zipRepositoryWithParticipation(Repository repo, String repositoryDir, boolean hideStudentName) throws IOException, UncheckedIOException {
+    public Path getRepositoryWithParticipation(Repository repo, String repositoryDir, boolean hideStudentName, boolean zipOutput) throws IOException, UncheckedIOException {
         var exercise = repo.getParticipation().getProgrammingExercise();
         var courseShortName = exercise.getCourseViaExerciseGroupOrCourseMember().getShortName();
         var participation = (ProgrammingExerciseStudentParticipation) repo.getParticipation();
 
-        // The zip filename is either the student login, team short name or some default string.
-        var studentTeamOrDefault = Objects.requireNonNullElse(participation.getParticipantIdentifier(), "student-submission" + repo.getParticipation().getId());
-
-        String zipRepoName = FileService.sanitizeFilename(courseShortName + "-" + exercise.getTitle() + "-" + participation.getId());
+        String repoName = FileService.sanitizeFilename(courseShortName + "-" + exercise.getTitle() + "-" + participation.getId());
         if (hideStudentName) {
-            zipRepoName += "-student-submission.git.zip";
+            repoName += "-student-submission.git";
         }
         else {
-            zipRepoName += "-" + studentTeamOrDefault + ".zip";
+            // The zip filename is either the student login, team short name or some default string.
+            var studentTeamOrDefault = Objects.requireNonNullElse(participation.getParticipantIdentifier(), "student-submission" + repo.getParticipation().getId());
+
+            repoName += "-" + studentTeamOrDefault;
+        }
+        repoName = participation.addPracticePrefixIfTestRun(repoName);
```

Comment:
I think this would make sense, too. However, I would say that changing the core domain object is out of scope for this PR.

Reply:
Because the result of `addPracticePrefixIfTestRun` is set into `repoName` again, I would not expect an update for the `participation`. Furthermore, there is a meaningful JavaDoc comment for the method. Only the `IfTestRun` part of the method name and the corresponding part of the JavaDoc is maybe a little outdated through 8a8f11c7af6d2cb33cb2a437f96812d02e6ba2c5, but `isPractiseMode` and `isTestRun` is the same, as stated in the JavaDoc of the first getter.

Beside the discussion on the quality of this methodname, @b-fein is absolutely right that this is out of scope for this PR. The only change in this line comes from the rename of `zipRepoName` to `repoName`.